news/2019-02-07-weld-310Final.asciidoc

```diff
@@ -0,0 +1,81 @@
+** `WeldManager`, providing extra methods over what `BeanManager` offers, is now an injectable bean (link:https://issues.jboss.org/browse/WELD-2538[WELD-2538, window="_blank"])
+** `WeldManager` now offers new util methods allowing you to easily grab active contexts
+*** This is mainly for the purpose of context propagation as it also allows you to grab all active contexts that support it
+** `WeldManager` can now be used to check if any given context is active without having to care about exceptions (link:https://issues.jboss.org/browse/WELD-2537[WELD-2537, window="_blank"])
+** Added new SPI for class defining, deprecated old approach in `ProxyServices` (link:https://issues.jboss.org/browse/WELD-2556[WELD-2556, window="_blank"])
+*** Integrators will have to implement this new API to be able to operate on JDK 12+
+*** Weld now delegates class defining to integrators in order to avoid having to crack open `ClassLoader` methods
+*** Previously, integrators provided `ClassLoader` instance which Weld then used to invoke `defineClass()` methods
+*** We not ask integrators to invoke those methods themselves while providing them with all the necessary bits for doing so
+** Remove deprecated parts of API/SPI (link:https://issues.jboss.org/browse/WELD-2558[WELD-2558, window="_blank"])
+*** Lots of unused and long deprecated methods were removed, should have no effect on code as those weren't used anyway (most of them since Weld 3.0.0)
+
+* Weld Core
+** Weld will now log a `WARNING` if you try to register an invalid qualifier (link:https://issues.jboss.org/browse/WELD-2522[WELD-2522, window="_blank"])
+** You can now use `InterceptionFactory` with an interface as a parameter (link:https://issues.jboss.org/browse/WELD-2550[WELD-2550, window="_blank"], link:https://issues.jboss.org/browse/WELD-2533[WELD-2533, window="_blank"])
+*** This means the proxy class will be based off an interface which is by definition always proxyable
+*** You can therefore even supply an unproxyable implementation and it will still work
+*** Note that this is experimental feature with some limitations to it, see link:http://docs.jboss.org/weld/reference/latest/en-US/html_single/#_loosening_the_limitations_of_literal_interceptionfactory_literal[Weld docs] for more details
+** Correct atomic behaviour in `RequestContextController` (link:https://issues.jboss.org/browse/WELD-2536[WELD-2536, window="_blank"])
+** Fix rare race condition in `ConcurrentValidator` (link:https://issues.jboss.org/browse/WELD-2545[WELD-2545, window="_blank"])
+** Small correction to interceptor resulution when they have no bindings (link:https://issues.jboss.org/browse/WELD-2521[WELD-2521, window="_blank"])
+** Enforce consistent behaviour between `AnnotatedType` and `WithAnnotations` in regards to how default methods (link:https://issues.jboss.org/browse/WELD-2551[WELD-2551, window="_blank"])
```

Comment:
... in regards to how default methods - the sentence looks unfinished

Reply:
just the `how` is redundant